source/common/config/watch_map.cc

```diff
@@ -42,6 +42,17 @@ AddedRemoved WatchMap::updateWatchInterest(Watch* watch,
                       findRemovals(newly_removed_from_watch, watch));
 }
 
+AddedRemoved WatchMap::addToWatchInterest(Watch* watch,
+                                          const std::set<std::string>& add_these_names) {
+  std::vector<std::string> newly_added_to_watch(add_these_names.begin(), add_these_names.end());
+  std::vector<std::string> newly_removed_from_watch;
+  watch->resource_names_.insert(add_these_names.begin(), add_these_names.end());
+  std::set<std::string> additions = add_these_names;
+
+  return AddedRemoved(findAdditions(newly_added_to_watch, watch),
+                      findRemovals(newly_removed_from_watch, watch));
```

Comment:
won't this always be the empty set?

Reply:
`findRemovals(newly_removed_from_watch, watch)` will be always empty, maybe giving a empty set here is more appropriate.

Reply:
sorry, I've already remove `additions`.